apps/prolb/prolb.sh

```diff
@@ -0,0 +1,166 @@
+AZHPC_VMSIZE=$(curl -s -H Metadata:true "http://169.254.169.254/metadata/instance?api-version=2018-10-01" | jq -r '.compute.vmSize')
+export AZHPC_VMSIZE=${AZHPC_VMSIZE,,}
+AZHPC_CORES=`cat $PBS_NODEFILE | wc -l`
+
+get_ib_pkey()
+{
+    key0=$(cat /sys/class/infiniband/mlx5_0/ports/1/pkeys/0)
+    key1=$(cat /sys/class/infiniband/mlx5_0/ports/1/pkeys/1)
+
+    if [ $(($key0 - $key1)) -gt 0 ]; then
+        export AZHPC_IB_PKEY=$key0
+    else
+        export AZHPC_IB_PKEY=$key1
+    fi
+
+    export AZHPC_UCX_IB_PKEY=$(printf '0x%04x' "$(( $AZHPC_IB_PKEY & 0x0FFF ))")
+}
+
+get_ib_pkey
+
+source /etc/profile # so we can load modules
+
+module use /usr/share/Modules/modulefiles
+module use $AZHPC_APPS/modulefiles
+module load gcc-8.2.0
+module load ${AZHPC_APPLICATION}_${APP_VERSION}
+if [ "$APP_VERSION" = "2.5.1" ]; then
+    module load mpi/openmpi-4.0.1
+else
+    # hpcx need to be build with C++ bindings
+    module load hpcx-2.4.1
+    if [ -n "${PROFILE}" ]; then
+        export HPCX_IPM_DIR=${HPCX_DIR}/ompi/tests/ipm-2.0.6
+        export IPM_KEYFILE=${HPCX_IPM_DIR}/etc/ipm_key_mpi
+        export IPM_LOG=FULL
+        export LD_PRELOAD=${HPCX_IPM_DIR}/lib/libipm.so
+
+        BARRIER=0  # Barrier is important to have a clean picture of the MPI API commands
+        # BARRIER==1 doesn't works here
+        if [[ "$BARRIER" == "1" ]]; then
+            export IPM_ADD_BARRIER_TO_REDUCE=1
+            export IPM_ADD_BARRIER_TO_ALLREDUCE=1
+            export IPM_ADD_BARRIER_TO_GATHER=1
+            export IPM_ADD_BARRIER_TO_ALL_GATHER=1
+            export IPM_ADD_BARRIER_TO_ALLTOALL=1
+            export IPM_ADD_BARRIER_TO_ALLTOALLV=1
+            export IPM_ADD_BARRIER_TO_BROADCAST=1
+            export IPM_ADD_BARRIER_TO_SCATTER=1
+            export IPM_ADD_BARRIER_TO_SCATTERV=1
+            export IPM_ADD_BARRIER_TO_GATHERV=1
+            export IPM_ADD_BARRIER_TO_ALLGATHERV=1
+        fi
+    fi
+fi
+
+numa_domains="$(numactl -H |grep available|cut -d' ' -f2)"
+AZHPC_PPR=$(( ($AZHPC_PPN + $numa_domains - 1) / $numa_domains ))
+
+printenv
+
+echo "downloading case ${CASE}..."
+start_time=$SECONDS
+cp $CASE_DIR/$CASE $AZHPC_JOBDIR/$CASE
+end_time=$SECONDS
+download_time=$(($end_time - $start_time))
+echo "Download time is ${download_time}"
+
+# self is required by PROLB
+mpi_options+=" -mca btl self"
+
+# Use UCX
+mpi_options+=" --mca pml ucx -mca osc ucx"
+mpi_options+=" -x UCX_NET_DEVICES=mlx5_0:1 -x UCX_IB_PKEY=$AZHPC_UCX_IB_PKEY -x UCX_LOG_LEVEL=ERROR"
+
+mpi_options+=" -x UCX_ZCOPY_THRESH=262144"
+
+# Enable HCOLL
+mpi_options+=" --mca coll_hcoll_enable 1 -x coll_hcoll_np=0 -x HCOLL_MAIN_IB=mlx5_0:1"
+
+mpi_options+=" --map-by ppr:$AZHPC_PPR:numa"
+mpi_options+=" -x LD_LIBRARY_PATH"
+mpi_options+=" -bind-to core"
+mpi_options+=" -report-bindings --display-allocation -v"
+
+if [ -n "${LD_PRELOAD}" ]; then
+    mpi_options+=" -x LD_PRELOAD"
+fi
+
+if [ "$AZHPC_VMSIZE" == "standard_hb60rs" ]; then
```

Comment:
this may need to be updated with a new SKU so might be good to add a note in the script and the readme to update this if needed. also a note to explain how we arrived at this number for these SKUs.

Reply:
Done in the readme and the run script